[WIP] Update suite to run against the latest

CaseyLeask opened the pull request:
There's a lot here I'm not clear on, like how to get Wattsi's full-page
version. After digging through
https://github.com/whatwg/html-build/blob/master/build.sh I have more of
an idea how this all sticks together, but there's so many different
toolchains it might take some time.

CaseyLeask committed:
WIP. Still getting anolis errors

There's a lot here I'm not clear on, like how to get Wattsi's full-page
version. After digging through
https://github.com/whatwg/html-build/blob/master/build.sh I have more of
an idea how this all sticks together, but there's so many different
toolchains it might take some time.

benschwarz commented:
We might want to update anolis and spec splitter etc etc? I guess by running make you've already got recent versions?

domenic commented:
So HTML doesn't use Anolis anymore. Wattsi is supposed to output a dev multipage version for you; you can see an older snapshot at https://html5.org/temp/multipage-dev.zip, or use Wattsi + build.sh to generate it directly.

CaseyLeask commented:
Thanks for the fast response folks! Wasn't expecting anything back for a bit.

To borrow from @benschwarz, is this the updated todo list?
- [ ] Update the suite to run against the latest dev multipage edition https://html5.org/temp/multipage-dev.zip or use Wattsi + build.sh to generate it directly
- [ ]  Publish it to gh-pages
- [ ]  Point DNS at github (@hixie)
- [ ]  Update design, modernise front-end
- [ ]  Investigate more modern tooling (python spec-splitter update, remove janky ruby, etc) OR, make moves to make the role of this repository to style and deploy the specification (rather than the insane processing etc that it does now).

CaseyLeask committed:
Used multipage output, removed splitter + anolis

CaseyLeask committed:
Getting something running

CaseyLeask commented:
I've reached the point where nothing crashes and it emits a zero error code on `make`. From this point on, I'll try my best to match the current output at https://developers.whatwg.org/

domenic commented:
That sounds about right, although some of the tooling might be obsolete, e.g. the spec comes pre-split. So maybe along the lines of your "OR".

On the other hand, if you get something working with updated content and auto-updating, that'll be a big improvement no matter the strategy :D

benschwarz commented:
Sounds good @CaseyLeask — I want to see the spec updated with the latest generated copy, then move to work on improving the tooling / generation / deployment, and then finally start working on the design and functionality of it.

benschwarz commented:
Thanks for the hard work so far @CaseyLeask :whale:

benschwarz commented on Rakefile:
```diff
@@ -71,19 +65,19 @@ namespace :postprocess do
+    #    end
+    #  end
+    #end
   end
```
Are we able to get these disabled lines figured out? Any questions?

CaseyLeask replied:
Currently it does this to the table of contents.

<img width="337" alt="screen shot 2015-12-31 at 4 14 09 pm" src="https://cloud.githubusercontent.com/assets/430277/12066086/99ad2640-afd9-11e5-8a25-6feda14f7412.png">

Need to investigate more.

CaseyLeask committed:
Updated Nokogiri

CaseyLeask committed:
Read all the html files, not just index

CaseyLeask committed:
Troubleshooting sections

CaseyLeask committed:
folded 'tidy' into the Rakefile

CaseyLeask committed:
Remove the second meta[charset]

We need to remove the extra charset specification, since we're adding
our own from html/head.html. We can't remove the <meta
http-equiv="Content-Type">, since there's a nokogiri bug coming from
libxml2 that just re-adds it, even when we specify a valid <meta
charset> https://github.com/sparklemotion/nokogiri/issues/1008

CaseyLeask committed:
Fix title in head

CaseyLeask committed:
Add link fixup script into head

CaseyLeask committed:
remove onload from body

CaseyLeask commented:
`head` is looking much better now. Onto the `body` of the doc.

CaseyLeask committed:
Add first header to docs

benschwarz commented:
:tada:

annevk commented:
@CaseyLeask let us know if you want us to generate something like  https://html5.org/temp/multipage-dev.zip automatically and ping your build script somehow. That might be easier than running a fork of the build script and such.

Filing an issue against https://github.com/whatwg/html-build once you're ready for that is probably easiest.

(Thanks @benschwarz for pointing me towards this thread!)

CaseyLeask commented:
@annevk That sounds like a good idea. I'm using https://html5.org/temp/multipage-dev.zip in this version.

I think I've still got a long way to go with this one. `spec-splitter` did a lot more wrangling than I gave it credit for.

annevk commented:
Note that we could also change https://github.com/whatwg/wattsi if the output for the developer edition is not great. Wattsi ultimately generates that so if there are things that are not usable we can simply change them.

CaseyLeask commented:
I'm keeping track of the issues in the branch, so I can keep a reasonable measure of progress.
https://github.com/CaseyLeask/developers.whatwg.org/issues

annevk commented:
@CaseyLeask let us know if things would be better to update upstream. Since we already generate a multipage developer copy, tweaking that might be a better solution than post-processing. And ensure things stay synchronized.

benschwarz commented:
@annevk, the biggest blocker is the lack of the TOC… a few other details, but that's the big one

annevk commented:
The multipage-dev that whatwg/html-build generates and I posted an output copy from at https://html5.org/temp/multipage-dev.zip has both a TOC for all documents in `index.html` and each individual page also has a TOC at the beginning. I checked just now and this is also true if you generate a new more up-to-date copy from scratch.

benschwarz commented:
@annevk Does it align with what's already deployed on developers.whatwg.org?

annevk commented:
I think that is definitely the intent, yes. It's basically the HTML standard without the sections marked as being specific to implementers (which we might not always annotate carefully since there's no current output to inspect, but we should fix that in the source document). I don't know how you got there prior to this point, but this seems like a good strategy.

CaseyLeask committed:
Cleanup a few messy parts

CaseyLeask committed:
Nicer formatted TOCs

CaseyLeask committed:
Removed debug logging

CaseyLeask committed:
Remove back and forward links

CaseyLeask committed:
Fixed titles for 'up-next' links

CaseyLeask committed:
A few up-next links are the 2nd child.

CaseyLeask committed:
Remove bottom nav links

benschwarz commented:
@CaseyLeask, I think we should merge what you have, then work on smaller specific issues from there, what's in master doesn't work at all, and this is at least half-way there. We can get it deploying to github pages etc, and start ironing out the kinks. 

Good with you?

CaseyLeask commented:
@benschwarz Good with me.

domenic commented:
Let us know when it's time to start setting up an automated system to deliver the dev version and deliver it to you.
